Tests/test_card_animation.gd

```diff
@@ -16,43 +16,57 @@ func after_each() -> void:
 
 
 func test_player_animation_created() -> void:
-	var card: CardBase = load("res://Cards/Resource/Card_Damage.tres").duplicate()
+	var card: CardBase = CardBase.new()
 	var cast_anim: CastAnimationData = CastAnimationData.new()
 	cast_anim.cast_animation_scene = load("res://Cards/Animation/Anim_Slap.tscn")
 	cast_anim.cast_position = CastPos_Caster.new()
+	
+	card.application_type = GlobalEnums.ApplicationType.ENEMY_ONLY
+	card.card_effects_data.append(EffectData.new())
 	card.card_effects_data[0].animation_data = cast_anim
+	card.card_effects_data[0].targeting_function = TargetingBase.new()
+	card.card_effects_data[0].effect = EffectDamage.new()
```

Comment:
any reason for creating the card this way instead of using the already made `Card_Damage.tres` ?

Reply:
I kept having a strange issue when trying to create a `Card_Damage` where it would affect other loaded `Card_Damage` resources from other tests causing the tests to fail. It did this even when calling `duplicate()` on the Resource. I couldn't figure it out so I just went with this

Reply:
oh I know this issue, it's a problem of loading cached resources
the solution is to force the card to be loaded without using the cache for the next test at least (or actually after any test that modifies the card)

Reply:
How do you force the card to be loaded?

Reply:
you need to do `load(path, cache_mode=2)`
here is [the documentation](https://docs.godotengine.org/en/stable/classes/class_resourceloader.html#enum-resourceloader-cachemode) on that

Reply:
Unfortunately this doesn't work for me. The problem arises with the `test_cast_position_all` test and `test_persist_strength_status` test. If I have both of these tests load `Card_Damage`, then the `test_persist_strength_status` test fails like this.
![image](https://github.com/Saplings-Projects/1M_sub/assets/151570126/0a5da00b-5b1a-454c-951c-a79c7f01f793)
I've found out it's because both tests use `_enemy_2`, which for some reason causes problems with the loaded card. If I replace the reference in `test_persist_strength_status` to `_enemy` instead of `_enemy_2` it works fine

Reply:
that's strange, the enemy shouldn't matter since they are built the same way iirc

Reply:
Well anyway let's leave it as is for now, trying to figure out something later on